internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala

```diff
@@ -376,15 +376,16 @@ object Incremental {
       }
     }
     val pickleJava = isPickleJava(scalacOptions)
+    val hasModified = initialInvClasses.nonEmpty || initialInvSources0.nonEmpty
     if (javaSources.nonEmpty && earlyOutput.isDefined && !pickleJava) {
       log.warn(
         s"-Ypickle-java should be included into scalacOptions if early output is enabled with Java sources"
       )
     }
     val initialInvSources =
-      if (pickleJava && initialInvSources0.nonEmpty) initialInvSources0 ++ javaSources
+      if (pickleJava && hasModified) initialInvSources0 ++ javaSources
```

Comment:
Can this be reproduced using either test like https://github.com/sbt/zinc/blob/develop/zinc/src/test/scala/sbt/inc/MultiProjectIncrementalSpec.scala or changes to one of scripted tests like https://github.com/sbt/zinc/tree/develop/zinc/src/sbt-test/source-dependencies/subproject-pipelining-mixed?